components/com_content/models/article.php

```diff
@@ -177,9 +177,13 @@ public function getItem($pk = null)
 				}
 
 				// Check for published state if filter set.
-				if ( !($haveToken == $currentToken) && ((is_numeric($published)) || (is_numeric($archived))) && (($data->state != $published) && ($data->state != $archived)))
+				if (((is_numeric($published)) || (is_numeric($archived))) && (($data->state != $published) && ($data->state != $archived)))
 				{
-					return JError::raiseError(404, JText::_('COM_CONTENT_ERROR_ARTICLE_NOT_FOUND'));
+					if( !($haveToken == $currentToken))
```

Comment:
This should be part of the if statement. As mentioned earlier, using the token in the database query, you don't have to keep doing the comparison. It then just becomes a true/false value.